tests/test_visitors/test_ast/test_loops/test_comprehensions/test_unsafe_generator_expression.py

```diff
@@ -62,11 +62,28 @@
 exponent = 10
 """
 
+unsafe_aug_value = """
+values = [1, 2, 3, 4, 5]
+exponent = 5
+power_expr = (value * exponent for value in values)
+exponent += 1
```

Comment:
What about changing `values` here? Like `values.append(1)`
Should we allow it?

Reply:
I think we should not because it affects the expected behavior of the expression. Can I create a new rule request to prevent it? 🤔

Reply:
I am not sure that it is "expected" because some people might be getting an error here. Let me give you an example:

```python
>>> values = [1, 2, 3]
>>> listcomp = [a * 2 for a in values]
>>> values.append(0)
>>> values, listcomp
([1, 2, 3, 0], [2, 4, 6])
```

Here's how it works with lists: no unexpected problems.
Now, the same with `gen`:

```python
>>> values = [1, 2, 3]
>>> gen = (a * 2 for a in values)
>>> values.append(0)
>>> values, list(gen)
([1, 2, 3, 0], [2, 4, 6, 0])
```

Now, we can see that changing `values` changes how `gen` works. It is not consistent with list comprehensions. If a persons just refactored list comprehension to be a generator expression with a code like this - there would be an error.

And we can catch it!

I recommend to try to visit `ast.Name` with usage / reassining context. It might be easier that our current appoach.

Reply:
Yes, I think that we should prevent it!

But, with the current visitor, it would be difficult to implement without making it complex because the `ast.Name` of values have `ctx` `ast.Load` in comparisons, functions or creating other list comprehensions and gens that not affect the `gen`.
```python
>>> values = [1, 2, 3, 4, 5]
>>> gen = (a * 2 for a in values)
>>> if 2 in values:
>>>    addition = sum(values)
>>> values, list(gen)
([1, 2, 3], [2, 4, 6]
```
Then we should check that the `ast.Attribute` append of `gen` is not being used within assignment and usage. That would need at least 2 new methods `visit_Attribute` and `_check_append` but the current visitor is at maximum allowed method capacity.

Also, I believe that a new violation would make the error easier to understand as the current violation only explains the variables that affect a generator expression. The `UnsafeAppendUsageViolation` would point out the forbidden usage of append within assignment and call for lists that affect a generator expression.

Reply:
Will the current implementation work for:

```python
self.x = 1
gen = (self.x * 2 for a in values)
self.x = 2
...
```

?

> That would need at least 2 new methods visit_Attribute and _check_append but the current visitor is at maximum allowed method capacity.

Don't worry about it for now, let's concentrate on making this work for all general and edge cases first, then we can refactor the working solution into a simplier version.

Reply:
Sure then, I'll be working on append support. 📝

On the other hand, the current implementation doesn't work with `self` variables. So, I'll update it as well with new tests 🤠